radix-transactions/src/validation/transaction_validator.rs

```diff
@@ -555,6 +590,23 @@ impl TransactionValidator {
         let non_root_subintents = &prepared.partial_transaction.non_root_subintents;
         let non_root_subintent_signatures = &prepared.non_root_subintent_signatures;
 
+        self.check_reference_limits(
```

Comment:
MINOR: Ideally this should live in `validate_v2_intent_cores_and_subintent_connection_counts` to prevent duplication between `validate_signed_partial_transaction_v2` and `validate_notarized_v2`

Reply:
See comments above - we do limit checks before validation.